actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java

```diff
@@ -963,6 +987,25 @@ private void sendEmailAndSms(Map<String, Object> userMap) {
     tellToAnother(EmailAndSmsRequest);
   }
 
+  private Response sendResetPasswordLink(Map<String, Object> userMap) {
+    Request passwordResetRequest = new Request();
+    if (StringUtils.isNotEmpty((String) userMap.get(JsonKey.EMAIL))) {
+      userMap.put(JsonKey.TYPE, JsonKey.EMAIL);
+    } else {
+      userMap.put(JsonKey.TYPE, JsonKey.PHONE);
+    }
+    passwordResetRequest.getRequest().putAll(userMap);
+    ProjectLogger.log(
+        "UserManagementActor: sendResetPasswordLink: sending reset password link to: "
+            + passwordResetRequest.getRequest().get(JsonKey.USER_ID),
+        LoggerEnum.INFO.name());
+    Response response =
+        (Response)
+            interServiceCommunication.getResponse(
```

Comment:
This is a bad idea - interServiceCommunication within the actor systems. Please get rid of it and call the actor directly.

Reply:
@indrajra if we want to call a actor from another actor, we are calling tellToAnother() in the BaseActor, but whatever actor we call from this is purely async calls. Here in my scenario sync call is needed so tellToAnother() cannot be used. Even InterServiceCommunicationImpl.getResponse also calling the actor the same way you suggested. Correct me if i am wrong.